packages/node/src/indexer/poi.service.ts

```diff
@@ -36,24 +27,69 @@ export class PoiService implements OnApplicationShutdown {
     this.isShutdown = true;
   }
 
-  async getLatestPoiBlockHash(schema?: string): Promise<string | Buffer> {
-    if (schema) {
-      this.poiRepo = PoiFactory(this.sequelize, schema);
-      const lastPoi = await this.poiRepo.findOne({
-        order: [['createdAt', 'DESC']],
-      });
-      if (lastPoi === null) {
-        this.latestPoiBlockHash = DEFAULT_PARENT_HASH;
-      } else if (lastPoi !== null && lastPoi.hash) {
-        this.latestPoiBlockHash = lastPoi.hash;
+  async init(schema: string): Promise<void> {
+    this.schema = schema;
+    const latestPoiBlockHash = await this.fetchLatestPoiBlockHashFromDb(schema);
+    this.latestPoiBlockHash = latestPoiBlockHash
+      ? latestPoiBlockHash
+      : DEFAULT_PARENT_HASH;
+    this.blockOffset = await this.fetchBlockOffsetFromDb(schema);
+  }
+
+  async fetchLatestPoiBlockHashFromDb(
+    schema: string,
+  ): Promise<string | Buffer | null> {
+    this.poiRepo = PoiFactory(this.sequelize, schema);
+    const lastPoi = await this.poiRepo.findOne({
+      order: [['createdAt', 'DESC']],
+    });
+    if (lastPoi === null) {
+      return null;
+    } else if (lastPoi !== null && lastPoi.hash) {
+      return lastPoi.hash;
+    } else {
+      throw new Error(`Poi found but can not get latest hash`);
+    }
+  }
+
+  async fetchBlockOffsetFromDb(schema: string): Promise<number | null> {
+    this.metadataRepo = MetadataFactory(this.sequelize, schema);
```

Comment:
this.metadataRepo initialization should only happen once.
in the constructor() or init()